code/__DEFINES/combat.dm

```diff
@@ -284,3 +284,16 @@ GLOBAL_LIST_INIT(shove_disarming_types, typecacheof(list(
 
 /// Proceed with the attack chain, but don't call the normal methods.
 #define SECONDARY_ATTACK_CONTINUE_CHAIN 3
+
+//Autofire component
+/// Compatible firemode is in the gun. Wait until it's held in the user hands.
+#define AUTOFIRE_STAT_IDLE (1<<0)
+/// Gun is active and in the user hands. Wait until user does a valid click.
+#define AUTOFIRE_STAT_ALERT	(1<<1)
+/// Gun is shooting.
+#define AUTOFIRE_STAT_FIRING (1<<2)
```

Comment:
No support for sleeping component?

Reply:
i didnt see use for it on tg

Reply:
to toggle firearm mode
not everyone likes autofire
ping issues, or just wanting to conserve ammo

Reply:
you can right click to shoot normally